C# SDK change with Hub's certificate and operation tasks

anusapan opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description
Regenerated the SDK based off a new swagger and add/update the test cases for updated SDK.
[Updated swagger PR](https://github.com/Azure/azure-rest-api-specs/commit/9bc154a142ae39c3b900ba91043aabed1ef8e2f3)

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

andrew-buckley commented on NuGet.Config:
```diff
@@ -1,8 +1,7 @@
-     <add key="NugetOfficialV3" value="https://api.nuget.org/v3/index.json" />
-     <add key="Local" value="tools/LocalNugetFeed" />
-   </packageSources>
+    <add key="NugetOfficialV3" value="https://api.nuget.org/v3/index.json" />
```
Why this change?

andrew-buckley commented on src/SDKs/IotHub/IotHub.Tests/Helpers/IotHubTestUtilities.cs:
```diff
@@ -21,6 +21,22 @@ public class IotHubTestUtilities
         public static string DefaultResourceGroupName = "DotNetHubRG";
         public static string DefaultUpdateResourceGroupName = "UpdateDotNetHubRG";
         public static string EventsEndpointName = "events";
+        public static string DefaultCertificateIotHubName = "DotNetCertificateHub";
```
Might be better to programmatically generate this certificate.

anusapan replied:
OpenSsl is not compatible with netcoreapp1.1

andrew-buckley reviewed (approved):
You might want to also squash your commits before making the next PR.

D1v38om83r committed:
Added AzureModuleSDK tests (#3964)

* Fixed build errors

Attributes were duplicated in AssemblyInfo.cs

* Recorded AutomationModules tests

* Addressed PR comments

Addressed PR comments, changed polling delay for the test to run fast.

* Reverted version update and genereate assembly info property

shutchings committed:
Machine Learning Compute: Add optional deleteAll parameter (#3959)

* Add deleteAll query parameter

* Remove unnecessary package tags

ro-joowan committed:
[ADLA] - Adding Scope APIs/Properties, existing Catalog APIs, and existing Account APIs (#3960)

* [CODE GEN] - ADLA ACCOUNT

* [CODE GEN] - ADLA CATALOG

* [CODE GEN] - ADLA JOB

* [VERSION] - ADLA 3.1.2-preview to 3.2.3-preview

* [CHANGELOG] - ADLA Update

* Patching OData type for Job LIST

* Updating tests and adding SessionRecords

* Minor test revision

* Updating Catalog tests and SessionRecords

* Updating Catalog test and SessionRecord

* Adding metadata files from Code Generation

* Adding metadata files from Project Build

* [Second] Adding metadata files from Code Generation

Forgot about resource-manager code gen

* [Second] Adding metadata files from Project Build

It looks like the props file wasn't fully updated, so I ran msbuild
build.proj again before building the project via the IDE

ro-joowan committed:
[ADLS] - Adding existing Account APIs (#3961)

* [CODE GEN] - ADLS ACCOUNT

* [VERSION] - ADLS 2.3.0-preview to 2.3.3-preview

* [CHANGELOG] - ADLS Update

* Skipping flaky test

* Updating tests and adding SessionRecords

* Cleaning up comments in ADLS data-plane cs files

* Adding metadata files from Code Generation

* Adding metadata files from Project Build

kc786 committed:
Added two new operations for Management Consumption (#3972)

* Added two new operations for Management Consumption
1. Reservation Summaries
2. Reservation Details

* updated  *.csproj and AssemblyInfo.cs files with the new version of the SDK.

* Perfrom steps mentioned by shahabhijeet in comments
Added package release notes

* AzureSdk.RP.props file changes

yuanliu-ms committed:
SDK update for ADF ExecuteSsisPackageActivity (#3936)

* SDK update for ADF ExecuteSsisPackageActivity

* Update 1

* Update SDK for ExecuteSSISPackageActivity [Iteration2]

* Revert "Update SDK for ExecuteSSISPackageActivity [Iteration2]"

This reverts commit 4d8be60805ce082d565a22f2d20b8743e2629b84.

* Update SDK for ExecuteSsisPackageActivity [Iteration2]

* SDK update for ADF ExecuteSsisPackageActivity [Iteration3]

* SDK update for ExecuteSsisPackageActivity in ADF

* Regenerate SDK

* Use generate.cmd to generate SDK

* Add AzSdk.RP.props file

JCookit committed:
Changed assembly and package name to official name. (#3984)

hyonholee committed:
Adding Priority property and PUT operation to VmssVM. (#3989)

ro-joowan committed:
[ADLA - ADLS] - Non-functional, minor fixes (comments and changelog) (#3990)

* Test comment fix

* ADLS changelog fix

* Comment fixes

shahabhijeet committed:
Update README.md

Updating readme for building the repo.

irrogozh committed:
Generated SDK for Connection monitor APIs + tests (#3988)

* Generated SDK for Connection monitor APIs + tests

* Added metadata

* Regenerated SDK with the new version of Autorest

* Changes based on CR

pcostantini committed:
[CognitiveServices] LUIS Programmatic SDK (#3987)

* LUIS Programmatic SDK - Updates based on latest Spec

* Add documentation for custom classes

* Updates based on latest Spec changes and Tests

* Cognitive Services Language SDKs directory re-structure

* Model CS  Language projects similar to other CS projects

bitsforthought committed:
[Azure Search] Upgrade Management SDK to version 2.0 (#3993)

* [Azure Search] Adding back missing tools and fixing build warnings

* [Azure Search] Regenerate mgmt SDK with latest version of AutoRest

* [Azure Search] Updating session records for management SDK tests

* [Azure Search] Adding the management SDK to the list of support Nuget packages

* [Azure Search] Adding useful script for regenerating management SDK

The stock generate.cmd script takes way too many parameters to be useful
when iterating on a private Swagger spec.

* [Azure Search] Generating new code for PATCH services and adding tests

Also re-recording all session records.

* [Azure Search] Adding support for Operations API

Added new generated code, plus a scenario test and corresponding session
record.

* [Azure Search] Bumping major version number of management SDK

A major version bump is necessary because the behavior of
Services.CreateOrUpdate has changed. Specifically, service creation and
scaling is now synchronous instead of asynchronous. Also, there is new API
surface area that represents breaking changes from the .NET point of view
(i.e. -- new methods added to existing interfaces).

* Fixing generate.cmd to point to the correct azure-rest-api-specs branch

Since the refactor, the default branch is master, not current.

* [Azure Search] Regenerating management SDK from the latest spec

This includes some minor doc changes.

bitsforthought committed:
[Azure Search] Fixing management SDK package name (#3996)

shahabhijeet committed:
Enabling multiple headers for LRO (#4002)

* Refator LRO to support multiple headers in a requests
* Refactoring check errors. Fixes applied for failing tests for autorest
* Adding comments and code cleanup
* more documentation for LRO rules applicable for each operation
* Updating release notes

chandrasekarsrinivasan committed:
Updating to 2017-11-01 for Network SDK  (#3999)

* Updating version number for Network SDK

* generate Network SDK for 11-01

* Update release notes

kalyanaj committed:
Event Grid: Regenerated the SDK and re-recorded the tests based on the newer API version (#4000)

* Regenerated the SDK and re-recorded the tests based on the newer API version.

* Re-recorded the tests.

* Updated package release notes to reflect this set of changes.

* Re-recorded the tests.

vinayada1 committed:
Pstestfail (#4003)

* Update release notes

* Generated code

hbzhxying committed:
DataFactory: convert all securestring to secretbase in Linked Service (#3997)

* change all securestring to secret base in linked service in sdk

* update the changelog.md

* use generate.cmd

* update metedata

shahabhijeet committed:
Update Auth library to relax ADAL references as range (#4006)

hvermis committed:
Updating ADF SDK version for the release (#4007)

* Updating ADF SDK version for the release

* Updating ADF SDK version for the release

avneeshraiusit committed:
Fixing wrong discriminator value for AzureFabricCreationInput (#3998)

* Fixing wrong discriminator value for AzureFabricCreationInput
* add metadata files

sudhansu88 committed:
Resolve Merge Conflicts for generate.cmd under EntitySearch (#4011)

shahabhijeet committed:
Fixing Exception tests that had incorrect Asserts which will always s… (#4009)

* Fixing Exception tests that had incorrect Asserts which will always succeed

* Addinga way to compare URIs between requests

kalyanaj committed:
EventGrid Data plane SDK: Re-generating SDK to include event types (#4018)

* Regenerated the SDK and re-recorded the tests based on the newer API version.

* Re-recorded the tests.

* Updated package release notes to reflect this set of changes.

* Re-recorded the tests.

* Updated the SDK based on the latest data-plane API specifications.

* Updated version and release notes.

lebronJ committed:
Regenerate Face/ComputerVision SDK with latest spec and update the unit tests, separate Face/CV aligning with other Vision APIs. (#3977)

chinadragon0515 committed:
Enable Enterprise Edition and Custom Setup Scripts for SSIS-IR (#4001)

shahabhijeet committed:
udpating newly release CR.Azure to be used in the SDK repo (#4023)

jeffrey-ACE-zz committed:
Add new sdk for ACE provisioning GSM, the sdk can be used to do partner management (#4019)

* FW:add sdk for ACE provisioning team

* Add more test cases

* Updated Swagger file, so copy new generated code and update test cases too

* Swagger is updated, so regenerated the SDK and update test cases

* Update namespace from gsm to managementpartner

* Renamed gsm to managementpartner in azure-rest-api-specs, so regenerated the sdks

vrdmr committed:
3.0.1-preview release: Contains SoftwareUpdateConfiguration resources API and newer tests. (#3992)

shihying committed:
Onboard API version 2017-08-01 (#4014)

* test1

* update tests

* scenario records

* fix warnings

* fix warnings

* add release note

* add metadata files

shihying committed:
Regenerate test file by record mode (#4028)

* test1

* update tests

* scenario records

* fix warnings

* fix warnings

* add release note

* add metadata files

* test files

shahabhijeet committed:
revert Azure.CR to previous version to unblock RP from publishing (#4033)

hyonholee committed:
Adding WriteAcceleratorEnabled property for Disks, adding ComputerName, OsName and OsVersion properties for VM instance view, and adding ForceRecoveryServiceFabricPlatformUpdateDomainWalk API for VM scale set. (#4029)

kosta-bizetic committed:
Added support for Autotuning API (#4015)

* Added support for Autotuning API

* Metadata change in sql_resource-manager.txt

azhao155 committed:
sdk generation related to Swagger pr: https://github.com/Azure/azure-… (#4026)

* sdk generation related to Swagger pr: https://github.com/Azure/azure-rest-api-specs/pull/2364

* generate from local build

* update to right version

bitsforthought committed:
[Azure Search] Fixing some long-lived bugs in the data plane SDK (#4036)

* [Azure Search] Restoring build script and bumping data plane SDK version

The version bump is in preparation for a series of bug fixes that have been
pending for a long time. The build script change was taken from the old
abandoned vs17Dev branch.

* [Azure Search] Fix for data plane SDK issue: 3309

The FieldBuilder class in the Search .NET SDK now ignores properties marked
with JsonIgnore.

* [Azure Search] GetClient propagates SearchServiceClient's HttpClientHandler

Fixes an issue whereby setting the HttpClientHandler when creating a
SearchServiceClient has no effect on the SearchIndexClient returned by
Indexes.GetClient. This would break scenarios such as enabling GZip
compression, for example.

Now the HttpClientHandler propagates, but the DelegatingHandlers don't since
they can't be cloned and are mutated by the ServiceClient base class
constructor.

hyonholee committed:
Adding LogAnalytics API. (#4035)

hbzhxying committed:
Azure data factory: Add AAD support for sql dw and azure sql (#4044)

bitsforthought committed:
[Azure Search] Remove JsonConvert.DefaultSettings check from unit tests (#4053)

Some time ago, we added a check to every Azure Search unit test that uses the
Run() method. This check is intended to ensure that none of our SDK code uses
the global JsonConvert.DefaultSettings, since these can be altered by other
code in the same AppDomain. This was causing issues for customers, and is the
reason we added SafeJsonConvert to ClientRuntime and AutoRest.

The check in the Run() method installs a custom JsonConverter and
ContractResolver that throw exceptions whenever they're called. It does this
using a try/finally block so that the change should only be visible while
running the unit test.

Unfortunately, since JsonConvert.DefaultSettings is global state, this
approach is fragile. For example, running tests in parallel can break this
because some infrastructure such as the VS test runner framework itself uses
JSON.NET in ways that rely on DefaultSettings.

Given the fragility of this approach, we've decided to remove these checks
altogether since they were causing spurious test failures.

kentzhang committed:
net sdk for consumption api v2018-01-31 (#4038)

* budget and marketplace api

* new generated code for consumption api

* fix subscription usage tests

* budget tests

* fix reservation tests

* re-generate sdk and update budget tests

* add budget tests for resource group name

* re-generate consumption sdk based on new swagger

* update consumption sdk test artifact

* marketplace tests pricesheet tests

* increase major version for consumption net sdk

* tests session records file update for api version bump up

* update package release notes

ro-joowan committed:
[ADLA] Account APIs object refactor and adding paging extension to Catalog ACL APIs  (#4043)

* [CODE GEN] Account

- Object refactor

* [CODE GEN] Job

* Changing OData Query for Job_List from JobInformation to
JobInformationBasic

* [CODE GEN] Catalog

* Making List_Acl pageable

* [CODE GEN] Metadata

* Updating tests

* Updating tests

* Updating SessionRecords

* [VERSION] 3.2.3-preview to 3.3.0-preview

* [ADLA] CODE GEN -- object refactor

* [CODE GEN] Metadata

* Fixing Account custom code

* [ADLA] - Partial revert - Global parameter to method parameter

* [CODE GEN] Metadata

* Updating tests

* [ADLA] Partial revert for some job APIs

* [CODE GEN] Metadata

* Updating SessionRecords

* [CHANGELOG] Update

* [CODE GEN] Metadata

* Fixing file name manually via git mv

Commit:
adding dns name label support for azure container instance .net sdk (#4049)

* adding dns name label support for azure container instance .net sdk

* update version number

* addressing the comments

* adding the missing metadata file

shahabhijeet committed:
fixing timeout issues for scenarios that sends retry-timeout (#4057)

* adding a way to detect test/playback mode. Enabled Retry-After for test mode. Now it is possible to add tests for Retry-After

* updating CR version for CR test projects

montgomp committed:
Device Provisioning Service (#4050)

* Recreate fork due to multiple issues in Azure/azure-sdk-for-net

* Update upstream per shahabhijeet.

Can you do the following in the below order:
Pull latest from upstream (if you have already done this good)
Execute msbuild.exe build.proj from the repo root
a. This will update the build tools in the current repo
Generate your SDK
Build you sdk at least one from command line of IDE
a. Msbuild.exe build.proj /t:build /p:Scope=SDKs\ProvisioningServices
Update the PR

* PR review requested changes

* Fix typo

* Update to handle if unexpected access policy is used.

* Feedback from shahabhijeet

* Remove AzSdk.RP.props file randomly created

* Regenerate client from latest swagger spec and rerecord tests.

* swagger spec renamed from "provisioningservice" to "deviceprovisioningservice"

* provisioningservices cleanup

* Feedback from @shabhijeet

zhangyd2015 committed:
Add two properties into LinkedIntegrationRuntime and update to latest… (#4041)

* Add two properties into LinkedIntegrationRuntime and update to latest spec

* Regenerate SDK according the comments

* Update change logs

ro-joowan committed:
[ADLS] Account APIs object refactor and couple of bugfixes (#4046)

* [CODE GEN] ADLS

* Removed ErrorDetails.cs -- it was unused
* Account to Accounts for operations
* Explicit parameter objects for creation and update

* [CODE GEN] Metadata

* Fixing custom code for file rename

* Updating tests

* Account refactor
* Permission field of AclStatus from int to string

* Updating tests

* Updating SessionRecords

* [VERSION] 2.3.3-preview to 2.4.0-preview

* [CHANGELOG] Update

* ADLS metadata update

* ADLA metadata update

bitsforthought committed:
[ClientRuntime] Fixing spurious test failures in JsonSerializationTests (#4060)

Two of the JsonSerializationTests modify JsonConvert.DefaultSettings. They do
this in order to validate that SafeJsonConvert does not rely on these
settings.

Before this change, these tests would install a custom JSON converter and
contract resolver that always throw. Unfortunately, other aspects of the test
infrastructure rely on DefaultSettings, so this would sometimes result in
unexpected exceptions being thrown.

The solution is to weaken the effect of changing DefaultSettings. Now, the
SafeJsonConvert tests install a custom converter for their own Model class
that produces incorrect results. Because it's targeted at that class
specifically, it should have no side effects on other code that relies on
DefaultSettings.

bitsforthought committed:
[Azure Search] Merge 4.x preview code to psSdkJson6 branch and bump version (#4062)

* Search SDK: Re-applying changes from search-4x-preview branch to vs17Dev

I tried rebasing the search-4x-preview branch to vs17Dev, but the results were
very messy, so instead this is just a squash-merge of all the changes. This
should differ from search-4x-preview only in the project files and some minor
test changes to account for behavioral changes in the latest version of
ClientRuntime.

* Search SDK: Fixing failing tests

* Search SDK: Re-recording mocks for the preview SDKs

* Search SDK: Updating copyright dates

* Search SDK: Rename Management SDK project and update build script (#3236)

* Renaming Management.Search to Microsoft.Azure.Management.Search

The package and assembly names were accidentally changed during the VS2017
migration. This fixes it.

* Search SDK: Updating Build-SearchPackages script

* [Azure Search] Bumping preview SDK version to 4.0.2-preview

This new version includes some minor bug fixes that just shipped in the 3.0.5
GA SDK.

wenbof committed:
SDK Change for SAP ECC (#4063)

Eva-Xiao committed:
add new LinkedService for vertica and netezza, update googleBigQuery (#4070)

* add new connector Veritca and Netezza, update Google BigQuery

* add new connector Veritca and Netezza, update Google BigQuery

* update changelog

* resolve conflicts

* update changelog

rajshah11 committed:
ManagementGroups SDK (#4071)

* ManagementGroups SDK

* Minor changes

ro-joowan committed:
Fixing readme (#4066)

hvermis committed:
[DataFactory] Updating the SDK version and release notes for 0.5.0 before publishing (#4077)

* Updating ADF SDK version for the release

* Updating ADF SDK version for the release

* Fixing the version for ADF SDK and updating the release notes

* Updated the version in csproj and AssemblyInfo

anusapan committed:
Merge branch 'psSdkJson6' of https://github.com/Azure/azure-sdk-for-net into psSdkJson0717

kosta-bizetic committed:
Changed version of Microsoft.Azure.Management.Sql to 1.12.0-preview (#4048)

hyonholee committed:
Adding ZoneBalance and PlatformFaultDomainCount for vm scale set (#4069)

anusapan committed:
Merge branch 'psSdkJson6' of https://github.com/Azure/azure-sdk-for-net into psSdkJson0717

v-sodsou committed:
Content moderator update (#4082)

* Content moderator update

* modified screen text api

* Updated the version

mihymel committed:
Improving reliability of SQL ServerKey tests by adding a short wait between key vault creation and key creation (#4072)

areddish committed:
Add Custom Vision Training and Prediction projects
Add session recordings for tests

jaredmoo committed:
Added more convenient method to rename database (#4056)

* Add convenient Database.Rename methods

Fixes #4055

* Added test recording

* Fixed duplicate release note

* Move custom code into customization folder

shriku11 committed:
Skps sdk json6 (#4061)

* Added SDK changes for ActionGroup Changes

* added change log entry

* fixed the Nuget.Config

* Commented the unecessary line

* Skipped unecessary test

* Revrted the version change for Newtonsoft library.

* Fixed a comment from SDK team

* Update Monitor.Tests.csproj

* Changed the Assemblyinfo.cs to change the version.

kalyanaj committed:
EventGrid SDK: Removing the preview suffix from the version (#4081)

* Regenerated the SDK and re-recorded the tests based on the newer API version.

* Re-recorded the tests.

* Updated package release notes to reflect this set of changes.

* Re-recorded the tests.

* Updated the SDK based on the latest data-plane API specifications.

* Updated version and release notes.

* Update package version to 1.3.0 to remove the preview suffix.

* Fixed a typo in the release notes.

haricr committed:
Add new features to DataFactory SDK (#4084)

* Add new features
1. SecureOutput and BatchCount
2. Filter Activity
3. LInked Service Parameters
4. FilterActivity

* Update from running Generate.cmd

areddish committed:
Update version

avneeshraiusit committed:
Changes corresponding to new swagger PR for API:2018-01-10 (#4075)

* Changes corresponding to new swagger PR for API:2018-01-10

* Fixin tests and updating test jsons

* Updating nuget version to 2.0.0

* Regenerating changes corresponding to comments on swagger PR.

* Updating metadata files as per review comment.

* Addressing review comments

q3blend committed:
Changing customConfig to long and making it a required parameter (#4017)

* Regenerating using generate.cmd

* Updating minor nuget version for customsearch

shahabhijeet committed:
Updating link for test framework documentation

felixwa committed:
Revise notice.txt of Cognitive Services (#4074)

* Revise notice.txt

Revise notice.txt after PMs confirmed with CELA.

* Resolve review comments.

Resolve review comments.

prakharsharma10 committed:
.NET SDK and its tests for the 2017-10-12  API version (#4076)

* .NET SDK and its tests for the 2017-10-12  API version

* Bump the version and generate the meta data file

* Updated the version and the metadata

* Fixed copyright symbol

* Made the AssemblyVersion and AssemblyFileVersion to match the nuget version

alexeldeib committed:
[App Insights] Adding Data Plane SDK  (#4088)

* Adding initial AI Data Plane

* Updated generated code & added some metric & event extensions

* Added metric extension tests

* Tests & fixes for event extensions

* Updated AI Data Plane SDK

* Minor cleanup to SDK

* Fix typing of eventId, rows

* Fix tests

* Update directory structure; generate.cmd; solution + project files

* Fix solution project file

* regen from pssdkjson6

* add git commit

* adding generated artifacts

* updating csproj, adding build artifacts

Commit:
ServiceBus:  properties added to queue and subscription (#4097)

* EnableBatchedOperations property added to queue, DeadLetteringOnFilterEvaluationExceptions added to Subscription

* bump the minor version

* version changed to 1.1.0

* updated resource-manager.txt

Commit:
[KeyVault] adding readme.md for data-plane (#4100)

* creating Azure KeyVault SDK readme for data-plane

* readme updates for PR feedback

Commit:
Update recommendation engine APIs and their test

Commit:
Update recommendation engine APIs and their test

Commit:
Generating SDK based on 2018-03-01-preview version.

payiAzure committed:
Generate sdk from swagger, add new test for cancel pool operation and list database/pool operation

hitenjava committed:
Update Azure Database Migration Service to add Managed Instance as target (#4098)

* Update generated files using specs from latest public preview swagger.

* Fix tests.

* Changes:
- Regenerate models to include commit id.
- Update minor version number.

* Add generated AzSdk.RP.props

* Change version number to 0.2.0

Commit:
Deleting old session records.

Commit:
Adding zone scenario tests for Private DNS.

Commit:
Modifying RecordSet tests for Private zones.

Commit:
Including session records again.

Commit:
Bumping up the version.

Commit:
Re-generating the SDK based from generate.cmd (fork: owolabi2/private-dns-public-preview).

Commit:
Merge branch 'psSdkJson6' of https://github.com/michimune/azure-sdk-for-net into psSdkJson6

Commit:
Fix assembly version and VersionPrefix, add .props file

darshanhs90 committed:
add dataaction attributes to permissions and Role Definition calls (#4079)

* update generated files

* update session recs and generated files

* fix failing tests

* comment extra line

* uncomment assemblyinfo file

* correct the session recs folder

* update session recs file to store and get ids

* add metadata files

* update csproj and assembleinfo file with newer version

* update props file

* update package notes

Commit:
Fix versions in .csproj files

bitsforthought committed:
[Azure Search] Upgrade data plane to latest AutoRest and split into 4 packages (#4104)

* [Azure Search] Upgrade data plane to the latest version of AutoRest

* [Azure Search] Splitting the data plane into four packages

What's motivating this change is a recognition that the data plane SDK is
really fulfilling three scenarios:

 1. Actually implementing search features in applications (full-text search,
    paging, faceted navigation, etc.)
 2. Indexing data via the push API
 3. Automating the management of indexes, synonyms, indexers, data sources,
    and other service-level resources.

Scenarios 1 and 2 are what you'd normally think of as "data plane". Scenario
3 is closer to "control plane" or management plane, or even "metadata plane"
since managing index schemas in particular is still pretty close to the data.

The split goes something like this:
 - Microsoft.Azure.Search.Data: Fulfills scenarios 1 and 2 above. This covers
   everything under the /docs resource. Currently most of this is custom code.
 - Microsoft.Azure.Search.Service: Fulfills scenario 3 above. This covers all
   top-level resources of the Azure Search Service REST API. This is mostly
   generated code, and our eventual goal is for nearly all of it to be
   generated.
 - Microsoft.Azure.Search.Common: Utilities common to Data and Service.
 - Microsoft.Azure.Search: A stub package that depends on all the others,
   plus defines the Indexes.GetClient() method that acts as a bridge from
   SearchServiceClient to SearchIndexClient. This package provides a
   convenient upgrade path for existing SDK users.

Eventually we're thinking of using Microsoft.Azure.Search.Service in
conjunction with Microsoft.Azure.Management.Search to implement a rich set
of PowerShell CmdLets that can manage indexes, data sources, indexers, etc.
in addition to the ARM-level resources.

* [Azure Search] Factoring out data plane assembly and package versions

Also bumping all data plane package major versions since moving types to
different assemblies is a breaking change.

* [Azure Search] Replacing ConverterBase with JsonReader extension methods

Reuse via inheritance is an anti-pattern, so we're getting rid of it.

* [Azure Search] Fixing issue with copying JsonSerializerSettings

Since the Azure .NET SDKs have to support a wide range of JSON.NET versions,
it isn't safe to reference properties of JsonSerializerSettings by name since
these properties come and go between versions. Instead, we use reflection to
dynamically generate code to copy a JsonSerializerSettings instance.

* [Azure Search] Versioning each data plane package separately

After some discussion, we decided it was better for the purposes of decoupled
development and release that we version each of the data plane packages
separately.

Commit:
Add web_resource-manager.txt

lorddev committed:
"automation" was misspelled (#4107)

Commit:
Bumping up the version again to 2.2.0 since 2.1.0 will be used by 2017-10-01 API version.

Commit:
Update AzureApiTag to AzSdk.RP.props and add SdkInfo_WebSiteManagementClient.cs

Commit:
Re-generating the SDK making the ZoneType optional.

pkamath13 committed:
[Monitor] SDK version 0.19.0-preview  (#4108)

* [Monitor] SDK version 0.19.0-preview

* [Monitor] SDK version 0.19.0-preview

* [Monitor] SDK version 0.19.0-preview

* [Monitor] SDK version 0.19.0-preview

* [Monitor] SDK version 0.19.0-preview

* [Monitor] SDK version 0.19.0-preview

felixwa committed:
initial commit - need to fix usage test case

initial commit - need to fix usage test case

bulentelmaci committed:
Add first version of .NET SDK for Policy Insights RP (#4117)

* Add SDK for PolicyInsights with initial set of tests (incomplete)

* Rest of the scope tests

* Minor changes

* Add query option tests

* Add policy insights to documentation files

* Changes requested as part of PR

* Fix generate.cmd
* Run generate to generate metadata
* Change version as suggested
* Remove hardcoded paths
* Add subscription ID to recorded session data instead of setting it in code
* Remove commented out code

solankisamir committed:
[Apim] Autorest based .NET client on 2018-01-01 version with Tests (#4115)

* 2018-01-01 client and basic tests

* Updated tests and generated expected metadata artifacts

solankisamir committed:
Fix ApiRevisionId and other test fixes

solankisamir committed:
Extension fixes

darshanhs90 committed:
update roledefinition filters (#4102)

* update tests to include type filter for roledefinitions

* update version

* rerecord tests

nkindberg committed:
Update DNS SDK for 2017-10-01 API Version (#4118)

* Update Azure DNS SDK for 2017-10-01 API Version

* Update test recordings for DNS SDK 2017-10-01 API Version

* Update 2017-10-01 DNS API version Swagger based on code review feedback

* Re-generate DNS SDK from 2017-10-01 official swagger

* Add AzSdk.RP.props for DNS SDK

Commit:
Including changes for 2017-10-01 API version.

Commit:
Re-recording all tests.

Commit:
Minor refactoring.

Commit:
Resolving merge conflicts.

Commit:
Fixing duplicated lines.

felixwa committed:
Regenerate to correct  limit/currentValue types

Regenerate with the latest swagger to correct  limit/currentValue types.
Now Usage UT can pass.

felixwa committed:
Bump version to 4.0.0.0

Bump version to 4.0.0.0

felixwa committed:
Update RecordSession for CognitiveServicesAccountUpdateWithCreateTest

Update RecordSession for CognitiveServicesAccountUpdateWithCreateTest

Commit:
Generating SDK against merged Swagger.

dsgouda committed:
Merge pull request #4120 from solankisamir/fix_apireviId

Fix ApiRevisionId and other test fixes

Commit:
capitalizing DNS in csproj

Commit:
Downgrading assemblyversion to 2.1.0.0

Commit:
* Add AzureDatabricks LinkedService and DatabricksNotebook Activity
  * Add headNodeSize and dataNodeSize properties in HDInsightOnDemand LinkedService.
  * Add LinkedService, Dataset, CopySource for SalesforceMarketingCloud

dsgouda committed:
Merge pull request #4109 from muwaqar/muwaqar/privatedns-public-preview

[do not merge before 4118] Adding changes to support Private DNS Public Preview

muwaqar committed:
Generating DNS client from latest Swagger changes

hyonholee committed:
Updating DiskRP API version to 2018-04-01 and adding zone resilient property to image storage profile (#4129)

payiAzure committed:
Sync the SDK repo with original. Sync official rest-api-specs with checked-in changes and renenerate the SDK

dsgouda committed:
Merge pull request #4130 from muwaqar/pdns-public-preview

Generating DNS client from latest Swagger changes

dsgouda committed:
Introducing generate.ps1 to replace generate.cmd (#4123)

* Adding generate.ps1

* better logging, hooks for generate.cmd

* Addressing comments

* Addressing PR comments

dsgouda committed:
Merge pull request #4116 from najian/psSdkJson6

add AzureDatabricks linkedservice and DatabricksNotebook activity

gucalder committed:
[Monitor] Fixing bug #2655: the top argument of the Metrics API must be a integer instead of a double

dsgouda committed:
Fixing remove-module error (#4133)

jaredmoo committed:
Update Auditing & ThreatDetection tests. (#4127)

Re-recorded tests with updated expecations due to minor changes in service behavior.

gucalder committed:
Merge branch 'psSdkJson6' of https://github.com/Azure/azure-sdk-for-net into psSdkJson6

payiAzure committed:
generate the .net sdk from official swagger repo azure/azure-rest-api-specs. Bump the api version in Assembly file and csproj

jeffrey-ace committed:
regenerate the sdk since the swagger is updated

gucalder committed:
[Monitor] Adding metadata file to PR

dsgouda committed:
Merge pull request #4110 from michimune/psSdkJson6

Fix recommendation engine APIs along with resource property fixes, add new APIs and tests

payiAzure committed:
Update ElasticPool test, only sleep when in Record mode

payiAzure committed:
Merge branch 'psSdkJson6' of https://github.com/Azure/azure-sdk-for-net into mybranch1

payiAzure committed:
do a git clean on working branch, rerun generate.cmd

payiAzure committed:
delete the release note for sdk version 1.12.0-preview

dsgouda committed:
Merge pull request #4128 from felixwa/MarchRelease

[Cognitive Services] March release (Add GetUsages operation, addnew/remove some API Kind)

dsgouda committed:
logging invoker information to file (#4137)

payiAzure committed:
fix build issue in ElasticPoolCrudScenarioTests.cs

hyonholee committed:
Add missing test for vmss redeploy and perform maintenance. (#4135)

hvermis committed:
Updating ADF SDK version and release notes before publishing (#4138)

* Update changelog.md

Updating the change log for release

* Update Microsoft.Azure.Management.DataFactory.csproj

Updating proj file with release notes

jeffrey-ace committed:
Merge branch 'psSdkJson6' of https://github.com/Azure/azure-sdk-for-net into psSdkJson6

dsgouda committed:
Merge pull request #4106 from payiAzure/mybranch1

Regenerating SDK based on swagger changes on List/Cancel operation for Azure database and Elastic pool

dsgouda committed:
Determining sdkdirectory for powershell invokers

trgrie committed:
Regen from swagger

trgrie committed:
LTR tests

dsgouda committed:
Merge pull request #4139 from dsgouda/generateps1

Determining sdkdirectory for powershell invokers

trgrie committed:
Adding tests that are actually recordable.

dsgouda committed:
Merge pull request #4134 from gucalder/psSdkJson6

[Monitor] Fixing bug #2655: the top argument of the Metrics API must be a integer instead of a double

trgrie committed:
Updating version info

trgrie committed:
Ran msbuild build.proj /t:Build /p:Scope=SDKs\SqlManagement per request.

jeffrey-ace committed:
Bump up managementpartner sdk version and change the package release notes

trgrie committed:
Undo version changes and add release notes for the newly added APIs and the breaking changes

gucalder committed:
[Monitor] Unblocking the publication of the SDK (#4141)

trgrie committed:
Address comments pertaining to tests

shahabhijeet committed:
Adding tests for all LRO Rules agreed upon with ARM team. (#4142)

* Adding tests for all LRO Rules agreed upon with ARM team. Accomodating 404, 204 for delete on Final GET. Accomodating text/plain content type certain RP still sending under pretext of application/json

* Updating E2E Scenario test

sudhansu88 committed:
Move Bing Search SDKs from Preview to GA

shahabhijeet committed:
Updating ClientRuntime release notes as well as version (#4145)

trgrie committed:
Regen from master

Regenerated from master, only the sql_resource-manager.txt is changed.

jeffrey-ace committed:
rerun generate.cmd to generate sdk

dsgouda committed:
Merge pull request #4140 from trgrie/psSdkJson6

LTR V2 .NET SDK Additions

jeffrey-ace committed:
Add generate.ps1 script and run it

dsgouda committed:
Fixing compute script and error handling fixes

jeffrey-ace committed:
Merge branch 'psSdkJson6' of https://github.com/Azure/azure-sdk-for-net into psSdkJson6

jeffrey-ace committed:
remove sdkDirectory argument in generate.ps1 script

dsgouda committed:
Merge pull request #4147 from dsgouda/generateps1

Removing output folder argument

dsgouda committed:
Adding instructions to generate code

sudhansu88 committed:
Address Comments

dsgouda committed:
Merge pull request #4136 from jeffrey-ACE/psSdkJson6

regenerate the sdk since the swagger is updated

dsgouda committed:
Merge pull request #4148 from dsgouda/docs

Adding instructions to generate code and open a PR

dsgouda committed:
instructions to generate code from custom repo

wilcobmsft committed:
1. Re-generate SDK
2. Update existing session records to reflect updated api-version

dsgouda committed:
reverting generate.cmd to old behavior

dsgouda committed:
Merge pull request #4150 from dsgouda/generateps1

reverting generate.cmd to old behavior

dsgouda committed:
Merge pull request #4149 from dsgouda/docs

instructions to generate code from custom repo

llidev committed:
Update SDK and test for BatchAI for GA

dsgouda committed:
Move Bing Search SDKs from Preview to GA

wilcobmsft committed:
Add enrollment account tests.

wilcobmsft committed:
Rev version.

llidev committed:
1. update SDK version 2. add PS generate script

wilcobmsft committed:
Fix versions.

wilcobmsft committed:
Add generate.ps1

llidev committed:
update release note for GA BatchAI SDKs

pavelzel committed:
WebSites ResourceHealthMetadata APIs and Tests

Adding the Microsoft.Web ResourceHealthMetadata APIs as well as new test cases for these APIs

pavelzel committed:
Updating WebSites SDK versions

Updating WebSites SDK versions

jaredmoo committed:
Regen from swagger

jaredmoo committed:
Fixed MaxSizeBytes string->long errors

jaredmoo committed:
Added customizations to improve backcompat

jaredmoo committed:
Fixed test build

jaredmoo committed:
Update TestLongTermRetentionV2Crud

wilcobmsft committed:
Minor test naming fixes.

wilcobmsft committed:
Re-generate using generate.ps1.

dsgouda committed:
[Billing RP] Add Enrollment Accounts APIs + tests.

[Billing RP] Add Enrollment Accounts APIs + tests.

xingwu1 committed:
Update Azure Batch .Net SDK to meet API version 2018-03-01.6.1. (#4151)

* Regenerate protocol based on 2018-02 Swagger spec

* Add object model layer for new APIs

* Update MSBuild version for ProxyLayerParser

* Add tests for new APIs

* Update Batch version to 8.1.0

 - Updated the changelog as well.

* Update to latest swagger spec.

* Add AzSdk.RP.props to source.

* Move LogUpload test case to IaaS pool.

areddish committed:
Update versions + generation scripts

dsgouda committed:
Fixing encoding so github can see diffs

shahabhijeet committed:
ReRecord failed test due to CR upgrade. (#4159)

* RPs whose tests have to be re recorded after CR update across the repo. Adding missing meta data files. Updating few test projects that have not been authored correctly

* ReRecording ApiManagement tests due to ClientRuntime update

* ApiMgmt, BatchMgmt and cdn rerecorded tests for CR update

* ReRecording Compute, Dns, MLCompute, RecSvcSiteRecovery, Sql, StreamAnalytics

* ReRecording DataFactory, EventHub, NotificaitonHub, Relay, StorSimple and TrafficManager tests due to CR change

* ReRecording ContainerRegistry, DataMigration, HDInsights, KeyVault, PowerBIDedicated, RedisCache, Resource, ServiceBus tests for CR Update

* ReRecording Customer Insights, EventGrid, MachineLearning test due to CR update.

* Fixing improper merge

* ReRecording Fabric.Admin tests due to CR update

shahabhijeet committed:
Merge branch 'psSdkJson6' into add-custom-vision

shahabhijeet committed:
Merge branch 'psSdkJson6' into psSdkJson6

llidev committed:
Merge branch 'psSdkJson6' into psSdkJson6

dsgouda committed:
Merge branch 'psSdkJson6' into generateps1

dsgouda committed:
Merge pull request #4161 from dsgouda/generateps1

Fixing encoding so github can see diffs

shahabhijeet committed:
Merge branch 'psSdkJson6' into add-custom-vision

llidev committed:
Update BatchAI test record

dsgouda committed:
Re-recording tests and fixes

dsgouda committed:
Merge pull request #4156 from pavelzel/psSdkJson6

WebSites ResourceHealthMetadata APIs and Tests

dsgouda committed:
Merge pull request #4152 from areddish/add-custom-vision

Add custom vision C# SDK

dsgouda committed:
Merge pull request #4162 from dsgouda/test

Re-recording tests and fixes

dsgouda committed:
Update SDK and test for BatchAI for GA

mjfrb committed:
subscription-2018-03-01-preview (#4157)

* subscription-2018-03-01-preview

* ran generate.ps1

* minor fixes

* reran tests

* added package release notes and bumped version

* pulled upstream changes, reran generate.ps1

* fix csproj

* fixed props

xingwu1 committed:
Update package information (#4163)

payiAzure committed:
generate sdk from swagger SQL database, elastic pool and capabilities 2017-10-01-preview. Re-recording all Sql tests

sarvesh-s committed:
restore point create and delete API missed test added (#4168)

Commit:
ServiceBus: 'properties' added to Rules CorrelationFilter (#4167)

* Added 'properties' in correlation filter

* updated session records

* code generated using generate.ps1

dsgouda committed:
adding license header to gen tool (#4173)

* adding license header to gen tool

* Update generateTool.ps1

hitenjava committed:
Update sdk with latest swagger spec.

murilogr committed:
Update generated data

murilogr committed:
Bump version

murilogr committed:
Update tests

Commit:
fixing xml comment on KeyWrapAlgorithm.CreateDecryptor

rohit-joy committed:
.Net SDK for PostgreSQL. (#4175)

* .Net SDK for PostgreSQL..

* Update test recordings.

* Resolve PR comments.

* Update props file and add metadata

hitenjava committed:
Update test recordings.

payiAzure committed:
re-record some failed tests due to subscription permission issue

dsgouda committed:
[KeyVault] Fixing typo in xml comment on KeyWrapAlgorithm.CreateDecryptor

hitenjava committed:
Handle PR feedback.

payiAzure committed:
fix tests. and re-recording all failed sdk tests

hitenjava committed:
Merge pull request #1 from Azure/psSdkJson6

Merge

murilogr committed:
Update release notes

hitenjava committed:
Update .props file

hitenjava committed:
Update .props file

hitenjava committed:
Fix .props file

murilogr committed:
Generate again with hotfix changes

murilogr committed:
Add generate.ps1

murilogr committed:
Regenerate with generate.ps1

murilogr committed:
Update props file

payiAzure committed:
fix restorepoint test using new database API and redo recording on this test

dsgouda committed:
Network 2018-02-01 release

dsgouda committed:
DMS - Add Agent Jobs and Login migration for SQL to MI public preview scenario

Commit:
Regenerated Commerce Admin

Commit:
Updated version and generate script

payiAzure committed:
Version bump for AssemblyInfo.cs and Management.Sql.csproj, and update the release note for new features of the new version

Commit:
Added generate.ps1, regenerated, including some text file

dsgouda committed:
AzureStack Commerce Admin SDK Regeneration

payiAzure committed:
sync with latest upstream/psSdkJson and regenerate sdk

payiAzure committed:
has build error, so regenerate and commit

v-sodsou committed:
Add SSN parameter to Screen Text API response

v-sodsou committed:
Change csproj version

dsgouda committed:
Content Moderator- Add SSN parameter to Screen Text API response

dsgouda committed:
Generate SDK.Net from swagger 'SQL database, elastic pool, and capabilities 2017-10-01-preview', and re-recording all Sql tests

anusapan committed:
Updated the SDK version

anusapan committed:
Merge branch 'psSdkJson6' of https://github.com/Azure/azure-sdk-for-net into psSdkJson0717
